File: my-app/src/api/superAdmin.ts
import { api } from "./client";

export interface AdminInfo {
  id: string;
  username: string;
  role: string;
  class_count: number;
  created_at: string;
}

export interface AdminListResponse {
  items: AdminInfo[];
}

export interface AdminCreateRequest {
  username: string;
  password: string;
}

export const superAdminApi = {
  listAdmins: () => api.get<AdminListResponse>("/super-admin/admins"),
  createAdmin: (data: AdminCreateRequest) =>
    api.post<AdminInfo>("/super-admin/admins", data),
};

File: backend/schemas/super_admin.py
import uuid
from datetime import datetime

from pydantic import BaseModel


class AdminResponse(BaseModel):
    id: uuid.UUID
    username: str
    role: str
    is_active: bool
    class_count: int
    category: str | None = None
    created_at: datetime


class AdminListResponse(BaseModel):
    items: list[AdminResponse]


class ResetPasswordRequest(BaseModel):
    new_password: str
